### PR TITLE
Flux Monitor V2 integration tests

### PR DESCRIPTION
**Pull request by jkongie**
Adds the following integration tests

* InvalidSubmission
* Deviation - Ported from `internal/features_test.go`
* NewRound - Ported from `internal/features_test.go` 
* HibernationMode
* AntiSpamLogic

### CODE REVIEW

File: core/internal/features_test.go [hunk 1]
```diff
@@ -790,7 +790,7 @@ func TestIntegration_FluxMonitor_Deviation(t *testing.T) {
 	kst.On("HasAccountWithAddress", address).Return(true)
 	kst.On("GetAccountByAddress", mock.Anything).Maybe().Return(accounts.Account{}, nil)
 	kst.On("SignTx", mock.Anything, mock.Anything, mock.Anything).Maybe().Return(&types.Transaction{}, nil)
-	kst.On("Accounts").Return([]accounts.Account{})
+	kst.On("Accounts").Return([]accounts.Account{{Address: address}})
```
Comment (connorwstein) [inline core/internal/features_test.go:793]: why the change here?

File: core/services/fluxmonitorv2/integrations_test.go [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+	fa.backend.Commit()
+
+	requestBody, err := json.Marshal(models.CreateJobSpecRequest{
+		TOML: string(s),
+	})
+	assert.NoError(t, err)
+
+	j := cltest.CreateJobViaWeb2(t, app, string(requestBody))
+
+	go func() {
```
Comment (connorwstein) [inline core/services/fluxmonitorv2/integrations_test.go:751]: I think we want to clean up this goroutine when the test exists no? i.e. 
```
	tick := time.NewTicker(1 * time.Second)
	defer tick.Stop()
	go func() {
		for range tick.C {
			b.Commit()
		}
	}()

```

File: core/services/fluxmonitorv2/integrations_test.go [hunk 3]
```diff
@@ -0,0 +1,946 @@
...
+			fa:              &fa,
+			roundId:         2,
+			answer:          int64(103),
+			from:            fa.nallory,
+			isNewRound:      true,
+			completesAnswer: true,
+		},
+		initialBalance-fee,
+		receiptBlock,
+	)
```
Comment (connorwstein) [inline core/services/fluxmonitorv2/integrations_test.go:471]: I think in this test it'd be good to add some basic assertions that the pipeline runs exist and look correct

File: core/services/fluxmonitorv2/integrations_test.go [hunk 4]
```diff
@@ -0,0 +1,946 @@
...
+		roundId:         roundId,
+		answer:          processedAnswer,
+		from:            fa.neil,
+		isNewRound:      true,
+		completesAnswer: false,
+	})
+
+	// - successfully close the round through the submissions of the other nodes
+	// Response by malicious chainlink node, nallory
+	initialBalance := currentBalance(t, &fa).Int64()
```
Comment (connorwstein) [inline core/services/fluxmonitorv2/integrations_test.go:805]: could use a comment about where this initialBalance comes from / represents - took us a little bit of time to figure out

File: core/internal/features_test.go (hunk 1 above)
Comment (jkongie) [reply core/internal/features_test.go:793]: I accidentally committed this change. I'll remove it

File: core/services/fluxmonitorv2/integrations_test.go (hunk 2 above)
Comment (jkongie) [reply core/services/fluxmonitorv2/integrations_test.go:771]: Fixed

File: core/services/fluxmonitorv2/integrations_test.go (hunk 3 above)
Comment (jkongie) [reply core/services/fluxmonitorv2/integrations_test.go:488]: Added an additional check to verify the pipeline run exists for the round, and verifies the value

File: core/services/fluxmonitorv2/integrations_test.go (hunk 4 above)
Comment (jkongie) [reply core/services/fluxmonitorv2/integrations_test.go:827]: Added a comment

**Review by connorwstein — APPROVED** [review]
